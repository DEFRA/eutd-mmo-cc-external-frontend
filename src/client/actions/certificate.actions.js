import { isError, showFullPageError } from './index';

export const GET_SUMMARY_DOCUMENT = 'GET_SUMMARY_DOCUMENT';
export const SHOW_SUMMARY_DOCUMENT_ERRORS = 'SHOW_SUMMARY_DOCUMENT_ERRORS';
export const GET_SUMMARY_UNAUTHORISED = 'GET_SUMMARY_DOCUMENT_UNAUTHORISED';

export function showSummaryCertificateErrors(data) {
  return { type: SHOW_SUMMARY_DOCUMENT_ERRORS, payload: { validationErrors: data } };
}

export const getSummaryCertificate = (journey, documentNumber = undefined) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? { headers: { documentnumber: documentNumber } }
    : {};

  await orchestrationApi.get(`/certificate/${journey}`, config)
    .then(
      res => {
        dispatch({ type: GET_SUMMARY_DOCUMENT, payload: res.data });
      },
      error => {
        if (isError(error)) {
          dispatch(showFullPageError());
        } else {
          dispatch(unauthorised());
        }
      });

    function unauthorised() { return { type: GET_SUMMARY_UNAUTHORISED }; }
};