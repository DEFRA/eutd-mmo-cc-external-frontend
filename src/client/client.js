// Startup point for the client side application
import 'babel-polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import {Router, withRouter} from 'react-router-dom';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import axios from 'axios';
import Routes from './Routes';
import reducers from './reducers';
import { createBrowserHistory } from 'history';

const composeEnhancers = process.env.NODE_ENV === 'development' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__  ?
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose;
  

const paramSerializerFn = params => {
  let output = '';
  Object.keys(params).forEach((k) => {
    output+= `${k}=${params[k]}&`;
  });
  return encodeURI(output);
};

const axiosResponseInterceptorSuccessFn = response => response;
const axiosResponseInterceptorErrorFn = error => {
  if (error.response.status === 599) {
    // Let's check if this is because we're logged out
    const empPromise = new Promise(() => {});
    const searchParam = error.request.responseURL.slice(error.request.responseURL.search(/[?]/g));
    let searchParams = new URLSearchParams(searchParam);
    if(searchParams.get('notLoggedInErr') === 'yes') {
      window.location = '/logout?backToPath=/server-logout';
      return empPromise;
    }
    // It's always a timeout...
    window.location = '/timed-out';
    return new Promise(() => {});
  } else {
    return new Promise((_,rej) => rej(error));
  }
};

const axiosInstanceForOrchService = axios.create({
  baseURL: '/orchestration/api/v1/',
  paramsSerializer: paramSerializerFn,
  timeout: window.INITIAL_STATE.config.axiosTimeout
});

axiosInstanceForOrchService.interceptors.response.use(axiosResponseInterceptorSuccessFn, axiosResponseInterceptorErrorFn);

const axiosInstanceForRefService = axios.create({
  baseURL: '/reference/api/v1/',
  paramsSerializer: paramSerializerFn,
  timeout: window.INITIAL_STATE.config.axiosTimeout
});

axiosInstanceForRefService.interceptors.response.use(axiosResponseInterceptorSuccessFn, axiosResponseInterceptorErrorFn);

const axiosInstanceForDynamixService = axios.create({
  baseURL: '/dynamix/',
  paramsSerializer: paramSerializerFn,
  timeout: window.INITIAL_STATE.config.axiosTimeout
});

axiosInstanceForDynamixService.interceptors.response.use(axiosResponseInterceptorSuccessFn, axiosResponseInterceptorErrorFn);

const store = createStore(
  reducers,
  window.INITIAL_STATE,
  composeEnhancers(applyMiddleware(
    // The difference between orch/ref service api is not significant as they both go through proxy
    // However it is significant in the server side store. To stick with same action creators we're
    // passing down same axios instance to both orchestration and reference service api .
    thunk.withExtraArgument( {
      orchestrationApi: axiosInstanceForOrchService,
      referenceServiceApi: axiosInstanceForRefService,
      dynamixApi: axiosInstanceForDynamixService } )
  ))
);

const ScrollToTop = withRouter( class ScrollToTop extends React.Component {
  componentDidMount() {
  }

  render() {
    return this.props.children;
  }
});

const history = createBrowserHistory();


ReactDOM.hydrate(
  <Provider store={store}>
    <Router history={history}>
      <ScrollToTop>
        <div>{renderRoutes(Routes)}</div>
      </ScrollToTop>
    </Router>
  </Provider>,
  document.querySelector('#root')
);
