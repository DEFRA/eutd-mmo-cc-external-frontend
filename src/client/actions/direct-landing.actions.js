import { beginApiCall, apiCallFailed, isForbidden, showFullPageError, CLEAR_ERRORS, isSilverLineError } from './index';
import { exportPayloadActionTypes } from '../actions';


export const GET_DIRECT_LANDING_PRODUCTS = 'get_direct_landing_products';
export const CLEAR_DIRECT_LANDING = 'clear_direct_landing';
export const DIRECT_LANDING_PRODUCTS_FAILED = 'direct_landing_products_failed';
export const POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'post_direct_landing_products_unauthorised';
export const GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'get_direct_landing_products_unauthorised';
export const POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR = 'post_direct_landing_prodcts_waf_error';

export const getDirectLandings =
  (documentNumber) =>
  async (dispatch, _getState, { orchestrationApi }) => {
    const config = { headers: { documentnumber: documentNumber } };

    try {
      dispatch(beginApiCall());

      const res = await orchestrationApi.get(
        '/export-certificates/export-payload/direct-landings',
        config
      );

      await dispatch({ type: GET_DIRECT_LANDING_PRODUCTS, payload: res.data });
    } catch (e) {
      if (isForbidden(e)) {
        dispatch({ type: GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
      } else {
        dispatch(apiCallFailed(e.response));
        throw new Error(
          `An error has occurred ${e.response.status} (${e.response.statusText})`
        );
      }
    }
  };

export const clearDirectLanding = () => ({ type: CLEAR_DIRECT_LANDING });

export const upsertDirectLanding = (landing, documentNumber = undefined) => (dispatch, getState, { orchestrationApi } ) => {
  const config = (documentNumber)
    ? {
        headers: {
          documentnumber: documentNumber
        }
    }
    : {};

  return orchestrationApi.post('/export-certificates/direct-landing/validate', landing, config)
    .then(
      res => {
        dispatch(landingUpserted(res.data));
        dispatch(clearErrors());
      },
      error => {
        if (error.response.status === 400) {
          dispatch(landingUpserted(error.response.data));
          throw new Error();
        } else if (isSilverLineError(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR, supportID: error.response.data.match(/\d+/)[0] });
        } else if (isForbidden(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
        } else {
          dispatch(showFullPageError());
        }
      });

  function landingUpserted(exportPayload) { return { type: exportPayloadActionTypes.LANDING_UPSERTED, exportPayload, errors: exportPayload.errors }; }
  function clearErrors() { return { type: CLEAR_ERRORS }; }
};
