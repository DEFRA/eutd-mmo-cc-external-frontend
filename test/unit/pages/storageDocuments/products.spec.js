import * as React from 'react';
import products from '../../../../src/client/pages/storageNotes/products';
import { getStorageNotesFromRedis, clearStorageNotes } from '../../../../src/client/actions';
import { component as ProductDetails } from '../../../../src/client/pages/storageNotes/products';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';

jest.mock('../../../../src/client/actions');

describe('loadData', () => {
  const store = {
    dispatch: () => {
      return new Promise((res) => {
        res();
      });
    }
  };

  const journey = 'storageDocument';
  const documentNumber = 'some-document-number';

  beforeEach(() => {
    getStorageNotesFromRedis.mockReturnValue({
      type: 'GET_STORAGE_NOTES_FROM_REDIS'
    });
  });

  it('will call all methods needed to load the component', async () => {
    products.documentNumber = documentNumber;

    await products.loadData(store, journey);

    expect(getStorageNotesFromRedis).toHaveBeenCalledWith(documentNumber);
  });
});

describe('Redirecting to Add product to this consignment page during loading', () => {
  const mockPush = jest.fn();
  const mockGetFromRedis = jest.fn();

  it('should redirect to add product to this consignment page when there is no catch', async () => {
    const props = {
      match: {
        params: { documentNumber: 'GBR-23423-4234234' }
      },
      global: {
        allFish: []
      },
      history: {
        push: mockPush,
      },
      exporter: {
        model: {
          exporterFullName: 'Mr Fish Exporter',
          exportCompanyName: 'MailOrderFish Ltd',
          addressOne: '5 Costa Road',
          townCity: 'Hackney',
          postcode: 'NE1 4FS',
          _updated: true
        }
      },
      storageNotes: {
        addAnotherProduct: 'notset',
        addAnotherStorageFacility: 'notset',
        catches: [],
        validationErrors: [{}],
        storageFacilities: [
          {
            facilityName: 'Test1',
            _facilityUpdated: false
          }
        ]
      },
      transport: {
        vehicle: 'plane',
        exportedTo: {
          isoCodeAlpha2: 'Alpha2',
          isoCodeAlpha3: 'Alpha3',
          isoNumericCode: 'IsoNumericCode',
          officialCountryName: 'Eritrea'
        }
      },
      fish: [{}],
      getFromRedis: mockGetFromRedis
    };

    await new ProductDetails.WrappedComponent(props).componentDidMount();

    expect(mockGetFromRedis).toHaveBeenCalledWith('GBR-23423-4234234');
    expect(mockPush).toHaveBeenCalledWith(
      '/create-storage-document/GBR-23423-4234234/add-product-to-this-consignment'
    );
  });
});

describe('Generate snapshot for the add products page', () => {
  const mockGetFromRedis = jest.fn();


  const props = {
    match: {
      params: { documentNumber: 'GBR-23423-4234234' }
    },
    global: {
      allFish: []
    },
    history: {
      location: {
        pathname: '/'
      }
    },
    exporter: {
      model: {
        exporterFullName: 'Mr Fish Exporter',
        exportCompanyName: 'MailOrderFish Ltd',
        addressOne: '5 Costa Road',
        townCity: 'Hackney',
        postcode: 'NE1 4FS',
        _updated: true
      }
    },
    storageNotes: {
      addAnotherProduct: 'notset',
      addAnotherStorageFacility: 'notset',
      catches: [],
      validationErrors: [{}],
      storageFacilities: [
        {
          facilityName: 'Test1',
          _facilityUpdated: false
        }
      ]
    },
    transport: {
      vehicle: 'plane',
      exportedTo: {
        isoCodeAlpha2: 'Alpha2',
        isoCodeAlpha3: 'Alpha3',
        isoNumericCode: 'IsoNumericCode',
        officialCountryName: 'Eritrea'
      }
    },
    fish: [{}],
    getFromRedis: mockGetFromRedis,
    route: {
      journey: 'storageDocument',
      previousUri: '/:documentNumber/you-have-added-a-product',
      progressUri: '/create-storage-document/progress'
    },
  };
  clearStorageNotes.mockReturnValue({
    type: 'CLEAR_STORAGE_NOTES'
  })
  const mockStore = configureStore([thunk]);

  it('should generate snapshot for the product page when there are 0 products', () => {
    const initialState = {
      storageNotes: {
        addAnotherProduct: 'notset',
        addAnotherStorageFacility: 'notset',
        catches: [],
        validationErrors: [{}],
        storageFacilities: [
          {
            facilityName: 'Test1',
            _facilityUpdated: false
          }
        ]
      }
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ProductDetails {...props} />
        </MemoryRouter>
      </Provider>
    );

      const { container } = render(wrapper);
      expect(container).toMatchSnapshot();
  });
  it('should generate snapshot for the product page when there is at least 1 product', () => {
    const initialState = {
      storageNotes: {
        addAnotherProduct: 'notset',
        addAnotherStorageFacility: 'notset',
        catches: [{
          product: "Narrownecked oceanic eel (ADD)",
          commodityCode: "11555555",
          certificateNumber: "123",
          productWeight: "11",
          weightOnCC: "11",
          placeOfUnloading: "Felix",
          dateOfUnloading: "01/01/2022",
          transportUnloadedFrom: "dss",
          scientificName: ""
      }],
        validationErrors: [{}],
        storageFacilities: [
          {
            facilityName: 'Test1',
            _facilityUpdated: false
          }
        ]
      }
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ProductDetails {...props} />
        </MemoryRouter>
      </Provider>
    );

      const { container } = render(wrapper);
      expect(container).toMatchSnapshot();
  });

  it('should generate snapshot for the product page when there are multiple products', () => {
    const initialState = {
      storageNotes: {
        addAnotherProduct: 'notset',
        addAnotherStorageFacility: 'notset',
        catches: [
          {
            product: 'Narrownecked oceanic eel (ADD)',
            commodityCode: '11555555',
            certificateNumber: '123',
            productWeight: '11',
            weightOnCC: '11',
            placeOfUnloading: 'Felix',
            dateOfUnloading: '01/01/2022',
            transportUnloadedFrom: 'dss',
            scientificName: '',
          },
          {
            product: 'Narrosecond oceanic eel (ADD)',
            commodityCode: '11555556',
            certificateNumber: '1234',
            productWeight: '11',
            weightOnCC: '11',
            placeOfUnloading: 'Felix',
            dateOfUnloading: '01/01/2022',
            transportUnloadedFrom: 'dss',
            scientificName: '',
          },
        ],
        validationErrors: [{}],
        storageFacilities: [
          {
            facilityName: 'Test1',
            _facilityUpdated: false,
          },
        ],
      },
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ProductDetails {...props} />
        </MemoryRouter>
      </Provider>
    );

      const { container } = render(wrapper);
      expect(container).toMatchSnapshot();
  });
});

describe('Back to progress page link', () => {
  let wrapper;
  beforeEach(() => {
    clearStorageNotes.mockReturnValue({ type: 'CLEAR_STORAGE_NOTES' });

    const mockStore = configureStore([thunk]);

    const props = {
      match: {
        params: {
          documentNumber: 'GBR-2021-CC-51F7BCC0A',
        },
      },
      route: {
        path: '/create-storage-document/123456789/add-product-to-this-consignment',
        previousUri:
          '/create-storage-document/:documentNumber/add-product-to-this-consignment',
        journey: 'storageNotes',
        progressUri: '/create-storage-document/progress',
        saveAsDraftUri: '/create-storage-document/storage-documents',
        title: 'Create a UK storage document - GOV.UK',
      }
    };

    const store = mockStore({
      clear: clearStorageNotes,
      global: {
        allFish: [],
      },
      errors: {},
      index: 0,
      scientificName: '',
      selectedDateOfUnloading: undefined,
      species: '',
      storageNotes: {
        addAnotherProduct: 'No',
        addAnotherStorageFacility: 'No',
        catches: [
          {
            certificateNumber: 'ABCD',
            commodityCode: '123456',
            dateOfUnloading: '14/10/2021',
            id: 'abcd-1634218473',
            placeOfUnloading: 'Dover',
            product: 'Atlantic cod (COD)',
            productWeight: '100',
            scientificName: '',
            transportUnloadedFrom: 'ABCD1234',
            weightOnCC: '100',
          }
        ],
        errors: {},
        errorsUrl: '',
        storageFacilities: [
          {
            _facilityUpdated: false,
            facilityAddressOne: 'MMO, LANCASTER HOUSE, HAMPSHIRE COURT',
            facilityBuildingName: 'LANCASTER HOUSE',
            facilityBuildingNumber: '',
            facilityCountry: 'ENGLAND',
            facilityCounty: 'TYNESIDE',
            facilityName: 'Some facility',
            facilityPostcode: 'NE4 7YH',
            facilityStreetName: 'HAMPSHIRE COURT',
            facilitySubBuildingName: 'MMO',
            facilityTownCity: 'NEWCASTLE UPON TYNE',
          }
        ],
        validationErrors: [{}],
      }
    });
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ProductDetails {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  it('should have a back to the progress page link', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper.find('a[href="/create-storage-document/progress"]').exists()).toBeTruthy();
  });
});

describe('should redirect to the forbidden page', () => {
  const mockPush = jest.fn();
  const mockGetFromRedis = jest.fn();

  it('should redirect to the forbidden page if storageNotes is unauthorised', async () => {
    const props = {
      match: {
        params: { documentNumber: 'GBR-23423-4234234' }
      },
      global: {
        allFish: []
      },
      history: {
        push: mockPush,
      },
      exporter: {
        model: {
          exporterFullName: 'Mr Fish Exporter',
          exportCompanyName: 'MailOrderFish Ltd',
          addressOne: '5 Costa Road',
          townCity: 'Hackney',
          postcode: 'NE1 4FS',
          _updated: true
        }
      },
      storageNotes: {
        unauthorised: true
      },
      transport: {
        vehicle: 'plane',
        exportedTo: {
          isoCodeAlpha2: 'Alpha2',
          isoCodeAlpha3: 'Alpha3',
          isoNumericCode: 'IsoNumericCode',
          officialCountryName: 'Eritrea'
        }
      },
      fish: [{}],
      getFromRedis: mockGetFromRedis
    };

    await new ProductDetails.WrappedComponent(props).componentDidUpdate();

    expect(mockPush).toHaveBeenCalledWith('/forbidden');
  });
});
