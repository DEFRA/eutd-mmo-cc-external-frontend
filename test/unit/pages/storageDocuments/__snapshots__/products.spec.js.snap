// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate snapshot for the add products page should generate snapshot for the product page when there are 0 products 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/orchestration/api/v1/storageNotes/back?n=/undefined/you-have-added-a-product"
          >
            Back
          </a>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            You have added 0 product to this consignment
          </h1>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/storageNotes/saveAndValidate"
        method="POST"
      >
        <h2
          class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
        >
          Product stored
        </h2>
        <br />
        <br />
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Do you need to add another product?
            </h2>
            <div
              class="e1vcssft0 css-1phgi8s-StyledFieldset-StyledHoc e4u1u2g0"
              id="addAnotherProduct"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              />
              <fieldset>
                <legend
                  class="visually-hidden"
                >
                  Do you need to add another product?
                </legend>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherProductYes"
                  id="label-addAnotherProductYes"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherProductYes"
                    name="addAnotherProduct"
                    type="radio"
                    value="Yes"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Yes
                  </span>
                </label>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherProductNo"
                  id="label-addAnotherProductNo"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherProductNo"
                    name="addAnotherProduct"
                    type="radio"
                    value="No"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    No
                  </span>
                </label>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <button
            class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
            formaction="/orchestration/api/v1/storageNotes/saveAndValidate?saveAsDraftUrl=undefined&c=undefined"
            id="saveAsDraft"
            name="saveAsDraft"
            type="submit"
          >
            Save as Draft
          </button>
          <input
            name="journey"
            type="hidden"
            value="storageDocument"
          />
          <input
            name="currentUri"
            type="hidden"
            value=""
          />
          <button
            class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
            id="continue"
            type="submit"
          >
            Save and continue
          </button>
        </div>
      </form>
      <a
        href="/create-storage-document/progress"
        id="backToProgress"
      >
        Back to Your Progress
        <span
          class="govuk-visually-hidden"
        >
          back to your progress page link
        </span>
      </a>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Generate snapshot for the add products page should generate snapshot for the product page when there are multiple products 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/orchestration/api/v1/storageNotes/back?n=/undefined/you-have-added-a-product"
          >
            Back
          </a>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            You have added 2 products to this consignment
          </h1>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/storageNotes/saveAndValidate"
        method="POST"
      >
        <h2
          class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
        >
          Product stored
        </h2>
        <dl
          class="css-39qxwj"
        >
          <div
            class="css-16k4qo5"
          >
            <dt
              class="css-a1pm7p"
            >
              Narrownecked oceanic eel (ADD)
            </dt>
            <span
              class="css-3ay1af"
            >
              <a
                class="css-80kuzi-StyledHoc e109g54z0"
                href="/create-storage-document/undefined/add-product-to-this-consignment/0"
                id="edit-product-0"
              >
                Edit
                <span
                  class="govuk-visually-hidden"
                >
                  product Narrownecked oceanic eel (ADD)
                </span>
              </a>
            </span>
            <span
              class="css-3ay1af"
            >
              <button
                class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
                formaction="/orchestration/api/v1/storageNotes/removeKey?n=/create-storage-document/undefined/you-have-added-a-product&key=product.0"
                id="remove-product-0"
                style="margin-left: 0.5em;"
              >
                Remove
                <span
                  class="govuk-visually-hidden"
                >
                  product Narrownecked oceanic eel (ADD)
                </span>
              </button>
            </span>
          </div>
        </dl>
        <dl
          class="css-39qxwj"
        >
          <div
            class="css-16k4qo5"
          >
            <dt
              class="css-a1pm7p"
            >
              Narrosecond oceanic eel (ADD)
            </dt>
            <span
              class="css-3ay1af"
            >
              <a
                class="css-80kuzi-StyledHoc e109g54z0"
                href="/create-storage-document/undefined/add-product-to-this-consignment/1"
                id="edit-product-1"
              >
                Edit
                <span
                  class="govuk-visually-hidden"
                >
                  product Narrosecond oceanic eel (ADD)
                </span>
              </a>
            </span>
            <span
              class="css-3ay1af"
            >
              <button
                class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
                formaction="/orchestration/api/v1/storageNotes/removeKey?n=/create-storage-document/undefined/you-have-added-a-product&key=product.1"
                id="remove-product-1"
                style="margin-left: 0.5em;"
              >
                Remove
                <span
                  class="govuk-visually-hidden"
                >
                  product Narrosecond oceanic eel (ADD)
                </span>
              </button>
            </span>
          </div>
        </dl>
        <br />
        <br />
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Do you need to add another product?
            </h2>
            <div
              class="e1vcssft0 css-1phgi8s-StyledFieldset-StyledHoc e4u1u2g0"
              id="addAnotherProduct"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              />
              <fieldset>
                <legend
                  class="visually-hidden"
                >
                  Do you need to add another product?
                </legend>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherProductYes"
                  id="label-addAnotherProductYes"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherProductYes"
                    name="addAnotherProduct"
                    type="radio"
                    value="Yes"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Yes
                  </span>
                </label>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherProductNo"
                  id="label-addAnotherProductNo"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherProductNo"
                    name="addAnotherProduct"
                    type="radio"
                    value="No"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    No
                  </span>
                </label>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <button
            class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
            formaction="/orchestration/api/v1/storageNotes/saveAndValidate?saveAsDraftUrl=undefined&c=undefined"
            id="saveAsDraft"
            name="saveAsDraft"
            type="submit"
          >
            Save as Draft
          </button>
          <input
            name="journey"
            type="hidden"
            value="storageDocument"
          />
          <input
            name="currentUri"
            type="hidden"
            value=""
          />
          <button
            class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
            id="continue"
            type="submit"
          >
            Save and continue
          </button>
        </div>
      </form>
      <a
        href="/create-storage-document/progress"
        id="backToProgress"
      >
        Back to Your Progress
        <span
          class="govuk-visually-hidden"
        >
          back to your progress page link
        </span>
      </a>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Generate snapshot for the add products page should generate snapshot for the product page when there is at least 1 product 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/orchestration/api/v1/storageNotes/back?n=/undefined/you-have-added-a-product"
          >
            Back
          </a>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            You have added 1 product to this consignment
          </h1>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/storageNotes/saveAndValidate"
        method="POST"
      >
        <h2
          class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
        >
          Product stored
        </h2>
        <dl
          class="css-39qxwj"
        >
          <div
            class="css-16k4qo5"
          >
            <dt
              class="css-a1pm7p"
            >
              Narrownecked oceanic eel (ADD)
            </dt>
            <span
              class="css-3ay1af"
            >
              <a
                class="css-80kuzi-StyledHoc e109g54z0"
                href="/create-storage-document/undefined/add-product-to-this-consignment/0"
                id="edit-product-0"
              >
                Edit
                <span
                  class="govuk-visually-hidden"
                >
                  product Narrownecked oceanic eel (ADD)
                </span>
              </a>
            </span>
            <span
              class="css-3ay1af"
            />
          </div>
        </dl>
        <br />
        <br />
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Do you need to add another product?
            </h2>
            <div
              class="e1vcssft0 css-1phgi8s-StyledFieldset-StyledHoc e4u1u2g0"
              id="addAnotherProduct"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              />
              <fieldset>
                <legend
                  class="visually-hidden"
                >
                  Do you need to add another product?
                </legend>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherProductYes"
                  id="label-addAnotherProductYes"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherProductYes"
                    name="addAnotherProduct"
                    type="radio"
                    value="Yes"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Yes
                  </span>
                </label>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherProductNo"
                  id="label-addAnotherProductNo"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherProductNo"
                    name="addAnotherProduct"
                    type="radio"
                    value="No"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    No
                  </span>
                </label>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <button
            class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
            formaction="/orchestration/api/v1/storageNotes/saveAndValidate?saveAsDraftUrl=undefined&c=undefined"
            id="saveAsDraft"
            name="saveAsDraft"
            type="submit"
          >
            Save as Draft
          </button>
          <input
            name="journey"
            type="hidden"
            value="storageDocument"
          />
          <input
            name="currentUri"
            type="hidden"
            value=""
          />
          <button
            class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
            id="continue"
            type="submit"
          >
            Save and continue
          </button>
        </div>
      </form>
      <a
        href="/create-storage-document/progress"
        id="backToProgress"
      >
        Back to Your Progress
        <span
          class="govuk-visually-hidden"
        >
          back to your progress page link
        </span>
      </a>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
